### PR TITLE
Added @NotNull annotations to improve Kotlin interoperability

### PR DESCRIPTION
**Pull request by bitmold**
This removes the ambiguous Kotlin _platform type_ which asserts that types returned from java are either not nullable or nullable. For example, calling `Ed25519PrivateKey.generate(SecureRandom)` from Kotlin would return an object that is of typed `Ed25519PrivateKey!` which is ambiguously either  `Ed25519PrivateKey` or the nullable `Ed25519PrivateKey?`. 

I went through the library and for all exposed methods that do not return null I've annotated them with `@NotNull`. None of the public methods could return null, but if they did it would be wise to annotate them in a similar fashion with `@Nullable` which informs the k-compiler to explicitly opt for the nullable version of the type. 

Additionally, parameters on public methods with object references that do not use null checks and would throw a NPE if called with `null` in java are also annotated with `@NotNull` which forces the Kotlin compiler to only accept non-null types on those methods and effectively prevents the library from throwing NPEs if used through Kotlin

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -23,14 +23,15 @@ sourceSets {
             // We need the entire source directory here, otherwise we get a
             // "package is empty or does not exist" error during compilation.
             srcDir("src/main/java")
+            include("org.jetbrains.annotations")
```
Comment (str4d) [inline build.gradle.kts]: Why is this necessary?

Comment (bitmold) [reply build.gradle.kts]: Hi, sorry to not get back to you right away! 

Honestly, I am not entirely sure this was necessary. I wasn't able to get the library to build without including this line; specifically the line `require annotations;` in `module-info.java` would fail. My knowledge on java 9 modules is very limited, so again I'm not sure this is necessary...

File: build.gradle.kts [hunk 2]
```diff
@@ -23,14 +23,15 @@ sourceSets {
...
             compileClasspath = sourceSets.main.get().compileClasspath
         }
     }
 }
 
 dependencies {
     implementation("cafe.cryptography:curve25519-elisabeth:0.1.0")
-
+    implementation("org.jetbrains:annotations:16.0.2")
```
Comment (str4d) [inline build.gradle.kts:34]: This library requires [JDK 8 or later](https://www.jetbrains.com/help/idea/annotating-source-code.html), but `ed25519-elisabeth` supports JDK 7. I'll re-evaluate at some point whether to drop JDK 7, but for now this should use `annotations-java5`.

File: src/main/java/module-info.java [hunk 3]
```diff
@@ -1,3 +1,4 @@
 module cafe.cryptography.ed25519_elisabeth {
     requires cafe.cryptography.curve25519_elisabeth;
+    requires annotations;
```
Comment (str4d) [inline src/main/java/module-info.java:3]:
```suggestion
    requires org.jetbrain.annotations_java5;
```
(see [here](https://github.com/JetBrains/java-annotations/blob/master/build.gradle#L43))

**Comment by str4d** [comment]
I've tested this branch in a Kotlin project and confirmed that it works for me.

File: build.gradle.kts (hunk 2 above)
Comment (str4d) [reply build.gradle.kts:34]: Oh wait, I mis-read the page. The backwards-compatible version is only required if the library is being compiled with a JDK prior to 8, but `ed25519-elisabeth` already requires JDK 9 or later to compile. So it's fine as-is.
